sdk/src/main/scala/io/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,270 @@
+          session)
+        log.error("Websocket On receive message processing exception occurred = " + ex.getMessage)
+    }
+  }
+
+  private def subscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+    val subscribeMethod = rpcParams.get(0)
+
+    subscribeMethod.asText() match {
+
+      case NEW_HEADS_SUBSCRIPTION.method =>
+        val subscriptionId = createSubscriptionId()
+        WebSocketAccountServerEndpoint.addNewHeadsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newHeads "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case NEW_PENDING_TRANSACTIONS_SUBSCRIPTION.method =>
+        val subscriptionId = createSubscriptionId()
+        WebSocketAccountServerEndpoint.addNewPendingTransactionsSubscription(Subscription(session, subscriptionId))
+        log.debug("New Subscription on newPendingTransactions "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case LOGS_SUBSCRIPTION.method =>
+        if (rpcParams.size() < 2)
+          WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+            new RpcError(RpcCode.InvalidParams, "Missing filters (address, topics).", "")),
+            session)
+        val subscriptionId = createSubscriptionId()
+        val filterQuery = EthJsonMapper.deserialize(rpcParams.get(1).toString, classOf[FilterQuery])
+        filterQuery.sanitize()
+        WebSocketAccountServerEndpoint.addLogsSubscription(SubscriptionWithFilter(session, subscriptionId,
+          filterQuery
+        ))
+        log.debug("New Subscription on logs "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case unkownMethod =>
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+          new RpcError(RpcCode.InvalidParams, "unsupported subscription type " + unkownMethod, "")),
+          session)
+        log.debug("unsupported subscription type " + unkownMethod)
+    }
+
+  }
+
+  private def unsubscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+
+    val subscriptionIdsToRemove: util.ArrayList[BigInteger] = new util.ArrayList[BigInteger]()
+    rpcParams.forEach(subscriptionId => {
+      subscriptionIdsToRemove.add(Numeric.decodeQuantity(subscriptionId.asText()))
+    })
+    val removedSubscription = WebSocketAccountServerEndpoint.removeSubscriptions(subscriptionIdsToRemove.toArray(new Array[BigInteger](0)))
+    if (!removedSubscription) {
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+        new RpcError(RpcCode.InvalidParams, s"Subscription ID not found.", "")),
+        session)
```

Comment:
Probably I miss understood your question but I think I've intended right since I replicated the same behavior of Eth 😄.
I've checked the Eth code and made some tests, It seems they accept an array of subscriptionsId but in reality they use only the first one.
Do we want to allow multiple unsubscriptions on a single call or stay strict to the Eth implementation?

Reply:
let's be like Geth: process the first one only and skip others.
But please, add the comment why it works in this way.

Reply:
Done